python/cuml/dask/datasets/make_regression.py

```diff
@@ -0,0 +1,216 @@
+        The number of parts of work.
+
+    Returns
+    -------
+    X : Dask-cuPy array of shape [n_samples, n_features]
+        The matrix.
+    """
+    rs = da.random.RandomState(seed=random_state,
+                               RandomState=cp.random.RandomState)
+    n = min(n_samples, n_features)
+
+    def generate_chunks_for_qr(total_size, min_size, n_parts):
+
+        n_total_per_part = max(1, int(total_size / n_parts))
+        if n_total_per_part > min_size:
+            min_size = n_total_per_part
+
+        n_partitions = int(max(1, total_size / min_size))
+        rest = total_size % (n_partitions * min_size)
+        chunks_list = [min_size for i in range(n_partitions-1)]
+        chunks_list.append(min_size + rest)
+        return tuple(chunks_list)
+
+    # Random (ortho normal) vectors
+    m1 = rs.standard_normal((n_samples, n),
+                            chunks=(generate_chunks_for_qr(n_samples,
+                                                           n, n_parts), -1))
+    u, _ = da.linalg.qr(m1)
+
+    m2 = rs.standard_normal((n, n_features),
+                            chunks=(-1, generate_chunks_for_qr(n_features,
+                                                               n, n_parts)))
+    v, _ = da.linalg.qr(m2)
+
+    # For final multiplication
+    n_samples_per_part = max(1, int(n_samples / n_parts))
+    u = u.rechunk({0: n_samples_per_part, 1: -1})
+    v = v.rechunk({0: n_samples_per_part, 1: -1})
+
+    # Index of the singular values
+    sing_ind = cp.arange(n, dtype=cp.float64)
+
+    # Build the singular profile by assembling signal and noise components
+    low_rank = ((1 - tail_strength) * cp.exp(-1.0 *
+                                             (sing_ind / effective_rank) **
+                                             2))
+    tail = tail_strength * cp.exp(-0.1 * sing_ind / effective_rank)
+    local_s = low_rank + tail
+    s = da.from_array(local_s,
+                      chunks=(int(n_samples_per_part),), asarray=False)
+
+    return da.dot(u * s, v)
+
+
+def make_regression(n_samples=100, n_features=100, n_informative=10,
+                    n_targets=1, bias=0.0, effective_rank=None,
+                    tail_strength=0.5, noise=0.0, shuffle=False, coef=False,
+                    random_state=None, n_parts=1):
+    """Generate a random regression problem.
+    The input set can either be well conditioned (by default) or have a low
+    rank-fat tail singular profile.
+
+    The output is generated by applying a (potentially biased) random linear
+    regression model with "n_informative" nonzero regressors to the previously
+    generated input and some gaussian centered noise with some adjustable
+    scale.
+
+    Parameters
+    ----------
+    n_samples : int, optional (default=100)
+        The number of samples.
+    n_features : int, optional (default=100)
+        The number of features.
+    n_informative : int, optional (default=10)
```

Comment:
Same question regarding defaults. Also not the biggest deal, but this values are pretty small for methods intended for multiple workers.